cmd/iut/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"runtime/debug"
+	"syscall"
+	"time"
+
+	config "github.com/eiffel-community/etos-api/internal/configs/iut"
+	"github.com/eiffel-community/etos-api/internal/iut/contextmanager"
+	"github.com/eiffel-community/etos-api/internal/logging"
+	server "github.com/eiffel-community/etos-api/internal/server"
+	"github.com/eiffel-community/etos-api/pkg/application"
+	"github.com/eiffel-community/etos-api/pkg/iut/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"github.com/snowzach/rotatefilehook"
+	"go.elastic.co/ecslogrus"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// main sets up logging and starts up the webserver.
+func main() {
+	cfg := config.Get()
+	ctx := context.Background()
+
+	var hooks []logrus.Hook
+	if fileHook := fileLogging(cfg); fileHook != nil {
+		hooks = append(hooks, fileHook)
+	}
+
+	logger, err := logging.Setup(cfg.LogLevel(), hooks)
+	if err != nil {
+		logrus.Fatal(err.Error())
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		logrus.Fatal(err.Error())
+	}
+	log := logger.WithFields(logrus.Fields{
+		"hostname":    hostname,
+		"application": "Dummy IUT Provider Service",
+		"version":     vcsRevision(),
+		"name":        "Dummy IUT Provider",
+		"user_log":    false,
+	})
+
+	// Database connection test
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{cfg.DatabaseURI()},
+		DialTimeout: 5 * time.Second,
+	})
+	if err != nil {
+		log.WithError(err).Fatal("failed to create etcd connection")
+	}
+
+	cm := contextmanager.New(cli)
```

Comment:
Do we really need to use the context manager here, there is not asynchronous proccessing of requests AFAICT.

Reply:
Removed in the most recent update.